server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java

```diff
@@ -823,6 +829,68 @@ public void getMailboxesShouldReturnMaxMessageQuotasForInboxWhenSet() throws Exc
             .body(FIRST_MAILBOX + ".quotas['#private&alice@domain.tld']['MESSAGE'].max", equalTo(43));
     }
 
+    @Test
+    @Ignore("Not handled by DefaultQuotaRootResolver for now")
+    public void getMailboxesShouldDisplayDifferentMaxQuotaPerMailboxWhenSet() throws Exception {
+        String mailboxId = mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, alice, DefaultMailboxes.INBOX).serialize();
+        String otherMailboxId = mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, alice, DefaultMailboxes.SENT).serialize();
+        quotaProbe.setMaxMessageCount("#private&alice@domain.tld&INBOX", QuotaCount.count(42));
+        quotaProbe.setMaxMessageCount("#private&alice@domain.tld&SENT", QuotaCount.count(43));
+
+        given()
+            .header("Authorization", accessToken.serialize())
+            .body("[[\"getMailboxes\", {\"ids\": [\"" + mailboxId + "\",\"" + otherMailboxId + "\"]}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxes"))
+            .body(ARGUMENTS + ".list", hasSize(2))
+            .body(FIRST_MAILBOX + ".quotas['#private&alice@domain.tld&INBOX']['MESSAGE'].max", equalTo(42))
+            .body(SECOND_MAILBOX + ".quotas['#private&alice@domain.tld&SENT']['MESSAGE'].max", equalTo(43));
+    }
+
+    @Test
+    public void getMailboxesShouldReturnQuotaRootForAllMailboxes() throws Exception {
+        given()
+        .header("Authorization", accessToken.serialize())
+        .body("[[\"getMailboxes\", {}, \"#0\"]]")
+    .when()
+        .post("/jmap")
+    .then()
+        .statusCode(200)
+        .body(NAME, equalTo("mailboxes"))
+        .body(ARGUMENTS + ".list*.quotas", new AllMatching<>(hasKey("#private&alice@domain.tld")));
+    }
+
+    class AllMatching<T> extends BaseMatcher<List<T>> {
```

Comment:
Deserves to be extracted from this class. 
Deserves to have a static factory method.

Reply:
done